src/libs/Pusher/pusher.ts

```diff
@@ -1,10 +1,36 @@
 import Onyx from 'react-native-onyx';
-import _ from 'underscore';
+import {Channel, ChannelAuthorizerGenerator, Options} from 'pusher-js/with-encryption';
+import isObject from 'lodash/isObject';
 import ONYXKEYS from '../../ONYXKEYS';
-import Pusher from './library';
+import Pusher from './library/types';
 import TYPE from './EventType';
 import Log from '../Log';
 
+type States = {
+    previous: string;
+    current: string;
+};
+
+type Args = {
+    appKey: string;
+    cluster: string;
+    authEndpoint: string;
+};
+
+type ChunkedDataEvents = {chunks: unknown[]; receivedFinal: boolean};
+
+type EventData = {id?: string; chunk?: unknown; final?: boolean; index: number};
+
+type SocketEventCallback = <T>(eventName: string, data?: T) => void;
+
+type PusherWithAuthParams = Pusher & {
+    config: {
+        auth?: {
+            params?: unknown;
+        };
+    };
+};
+
```

Comment:
```suggestion

type PusherEventName = LiteralUnion<DeepValueOf<typeof TYPE>, string>;
```

We'll need this one for the next functions.